types/styled-theming/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Type definitions for styled-theming 2.2
+// Project: https://github.com/styled-components/styled-theming#readme
+// Definitions by: Arjan Jassal <https://github.com/ArjanJ>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare function theme(name: string, values: theme.ThemeMap): theme.ThemeSet;
+
+declare namespace theme {
+    type ThemeValueFn = (props: object) => string;
+    type ThemeValue = string | ThemeValueFn;
+
+    interface ThemeMap {
+        [key: string]: ThemeValue;
+    }
+
+    interface VariantMap {
+        [key: string]: ThemeMap;
+    }
+
+    type ThemeSet = (props: object) => string;
+    type VariantSet = (props: object) => string;
+
+    function variants(
+        name: string,
+        prop: string,
+        values: VariantMap
+    ): VariantSet;
+}
+
+export default theme;
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`.

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Reply:
Thanks Daniel, I've made the change.